### DISCUSSION TITLE
Support for additional builtin cmake release build types

### DISCUSSION THREAD

**Discussion by djmittens** [jrouwe/JoltPhysics#1248] [Q&A] [answered]
So according to the [ docs here](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html)
 
There are a couple of possible build types availablet that are release but do not include the profiler nor the debug renderer.
I personally use `RelWithDebInfo` to build my project, and then turn debug mode on / off on individual files, that way i have symbols at least available for some things. 

Any chance to add those build types for jolt ?

https://github.com/jrouwe/JoltPhysics/blob/cede24d2733a4a473c6d486650ca9b6d0481681a/Jolt/Jolt.cmake#L563-L574

**Comment by jrouwe** [comment, marked as answer]
As far as I know, you can include Jolt's CMakeLists.txt file in your own CMakeLists.txt file. When you do, you can define any compile targets you like and override the defines in whatever way you like. I don't want to clutter the set of targets that I defined with extra (duplicate) ones, I think that will only add confusion.

**Reply by djmittens** [reply]
Fair enough. 
I found this behavior surprising when integrating this library with an existing project, as all the other libraries that use cmake system, did not leave out definitions in case of builtin build types. 
It took me a bit of time to figure this out, as the actual error i was getting is unexported typeinfo for one of the classes, which led me on an RTTI adventure.

Is the `JPH_DEBUG_RENDERER` definition considered private implementation detail for the project that may change in the future or is this something i can consider to  be stable part of the interface for using the library, and i can just assign it from my own build system without seeing a side effect?

**Reply by jrouwe** [reply]
`JPH_DEBUG_RENDERER` is a stable part of the API.
